app/api/schema/discount_codes.py

```diff
@@ -153,6 +154,55 @@ def validate_quantity(self, data, original_data):
                 raise UnprocessableEntity({'pointer': '/data/attributes/tickets-number'},
                                           "tickets-number should be greater than max-quantity")
 
+    @validates_schema(pass_original=True)
+    def validate_value(self, data, original_data):
+        if 'id' in original_data['data']:
+            try:
+                discount_code = DiscountCode.query.filter_by(id=original_data['data']['id']).one()
+            except NoResultFound:
+                raise ObjectNotFound({'parameter': '{code}'}, "DiscountCode: not found")
+
+            if 'type' not in data:
+                data['type'] = discount_code.type
+
+            if 'value' not in data:
+                data['value'] = discount_code.value
+
+        if data['type'] == "percent":
+            if 'tickets' in data:
+                for ticket in data['tickets']:
+                    ticket_object = Ticket.query.filter_by(id=ticket).one()
+                    if not ticket_object.price:
```

Comment:
shouldn't it be compared to zero ? for clarity?

Reply:
@CosmicCoder96 actually we are allowing none for price attribute of ticket. Though for front-end we send 0 in case ticket price is not set but server is also for android. Maybe there are tickets present with None as ticket price so it may cause discrepancies
`price = fields.Float(validate=lambda n: n >= 0, allow_none=True)`
I checked it just now. Should I compare explicitly?

Reply:
@shreyanshdwivedi okay, cool then don't change it.